### PR TITLE
GCI-<Nayan>:Done

### PR DESCRIPTION
**Pull request by NASA12345**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+// Database information inserted
+const (
+	DbUser     = "meclpohq"
+	DbPassword = "h6jPxvrJQkTmTYRZge1f0ZSShyEjItzI"
+	DbName     = "meclpohq host=rajje.db.elephantsql.com"
+	DbHost     = "rajje.db.elephantsql.com"
+)
+
+var router *gin.Engine
+var db *sql.DB
```
Comment (f4ww4z) [inline main.go]: You should be using `*sqlx.DB`, it has better and more concise methods.

File: main.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	}
+	defer db.Close()
+}
+
+func initDatabase() {
+
+	psqlInfo := fmt.Sprintf("user=%s password=%s dbname=%s host=%s", DbUser, DbPassword, DbName, DbHost)
+	var err error
+
+	db, err = sql.Open("postgres", psqlInfo)
```
Comment (f4ww4z) [inline main.go]: Use `db.MustConnect()` instead. You don't need to do `Open` and `Ping`.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+	err = db.Ping()
+	if err != nil {
+		panic(err)
+	}
+	log.Println("Successfully connected to the database")
+}
+// Provide details of all patients
+func GetAllPatients(c *gin.Context) {
+
+	rows, err := db.Query("SELECT * FROM patient")
```
Comment (f4ww4z) [inline main.go]: Use `*sqlx.DB#Select()` method instead, it's more readable.

File: main.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		}
+		patients = append(patients, row)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	c.JSON(http.StatusOK, gin.H{"patients": patients})
```
Comment (f4ww4z) [inline main.go]: No need to wrap it with "patients" attribute, just return the array of patients.

File: main.go [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+// Provide deatils of all locations
+func GetAllLocations(c *gin.Context) {
+	rows, err := db.Query("SELECT * FROM location")
```
Comment (f4ww4z) [inline main.go]: Use `*sqlx.DB#Select()` method instead, it's more readable.

File: main.go [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+		}
+		locations = append(locations, row)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	c.JSON(http.StatusOK, gin.H{"locations": locations})
```
Comment (f4ww4z) [inline main.go]: No need to wrap, just return the array of locations.

File: main.go [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+// Provoide details of all hospitals
+func GetAllHospitals(c *gin.Context) {
+	rows, err := db.Query("SELECT * FROM hospital")
```
Comment (f4ww4z) [inline main.go]: Use `*sqlx.DB#Select()` method instead, it's more readable.

File: main.go [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+		}
+		hospitals = append(hospitals, row)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	c.JSON(http.StatusOK, gin.H{"hospitals": hospitals})
```
Comment (f4ww4z) [inline main.go]: No need to wrap it, just return the array of hospitals.

**Review by f4ww4z — COMMENTED** [review]
Good start @NASA12345 , see my comments below.

File: main.go (hunk 1 above)
Comment (NASA12345) [reply main.go]: ok

File: main.go (hunk 2 above)
Comment (NASA12345) [reply main.go]: ok

File: main.go (hunk 3 above)
Comment (NASA12345) [reply main.go]: okk

**Comment by NASA12345** [comment]
Done

**Comment by NASA12345** [comment]
Please review my task on gci dashboard

**Review by f4ww4z — APPROVED** [review]
Looks good @NASA12345 !

**Comment by f4ww4z** [comment]
@NASA12345 please squash your commits next time.

**Comment by NASA12345** [comment]
Okk

**Pull request merged**